### PR TITLE
[FIX] mrp_byproduct: Don't allow same product as byproduct

### PR DESCRIPTION
**Pull request by sswapnesh**
Description of the issue/feature this PR addresses: Fixes https://github.com/odoo/odoo/issues/29808

Current behavior before PR: 

Desired behavior after PR is merged:  Bad configuration is not allowed
screenshot

Cc: @jco-odoo 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
@amoyaux @sle-odoo Makes sense.

**Comment by sswapnesh** [comment]
@amoyaux @sle-odoo Could you please review?

File: addons/mrp_byproduct/models/mrp_bom.py [hunk 1]
```diff
@@ -10,3 +11,13 @@ class MrpBom(models.Model):
...
 
     sub_products = fields.One2many('mrp.subproduct', 'bom_id', 'Byproducts', copy=True)
+
+    @api.constrains('product_id', 'product_tmpl_id', 'sub_products')
+    def _check_byproduct_recursion(self):
+        for bom in self:
+            if bom.product_id:
+                if bom.sub_products.filtered(lambda x: x.product_id == bom.product_id):
+                    raise ValidationError(_('Byproduct %s should not be same as BoM product.') % bom.display_name)
+            else:
```
Comment (sswapnesh) [inline addons/mrp_byproduct/models/mrp_bom.py]:
```suggestion
            elif:
```

**Comment by sswapnesh** [comment]
@amoyaux  Based on https://github.com/odoo/odoo/pull/33280#discussion_r282483309
Isn't it better to stop adding same product in BoM instead of Blocking Manufacturing order ? (Maybe it was to fix already created MOs)

Meanwhile I think we can close this as It is already Improved in https://github.com/odoo/odoo/pull/33280/commits/5b0d9038cabbd5c0a0b95d1896a5ea358013ca70

**Comment by amoyaux** [comment]
@sswapnesh It is for a stable version, so we couldn't add a constraint for existing database with the issue. It could be a good idea to remove the fix in master and add the constraints

**Comment by sswapnesh** [comment]
Can I go ahead and Propose this for master?( If there are no internal task for this)

**Comment by sswapnesh** [comment]
Ping @amoyaux

**Comment by sswapnesh** [comment]
Ping @sle-odoo  Is it still going to be merged or Maybe I should move this to master ?

**Comment by sle-odoo** [comment]
it seems it's already handled at https://github.com/odoo/odoo/commit/5b0d9038cabbd5c0a0b95d1896a5ea358013ca70 

since nobody complained about this issue and i'm wary of adding such changes in 12, let's keep it as it is

thank you

**Comment by sswapnesh** [comment]
Yes True but still there is no reason to allow the same Product as byproduct and Block Operation later on processing MO.


IMO we can remove 5b0d903 in master and block this on at Product level like this patch.

**Pull request closed without merging**